aws/resource_aws_lightsail_load_balancer_test.go

```diff
@@ -0,0 +1,305 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_lightsail_load_balancer", &resource.Sweeper{
+		Name: "aws_lightsail_load_balancer",
+		F:    testSweepLightsailLoadBalancers,
+	})
+}
+
+func testSweepLightsailLoadBalancers(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).lightsailconn
+
+	input := &lightsail.GetLoadBalancersInput{}
+	var sweeperErrs *multierror.Error
+
+	for {
+		output, err := conn.GetLoadBalancers(input)
+
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping Lightsail Load Balancer sweep for %s: %s", region, err)
+			return nil
+		}
+
+		if err != nil {
+			return fmt.Errorf("Error retrieving Lightsail Load Balancers: %s", err)
+		}
+
+		for _, loadBalancer := range output.LoadBalancers {
+			name := aws.StringValue(loadBalancer.Name)
+			input := &lightsail.DeleteLoadBalancerInput{
+				LoadBalancerName: loadBalancer.Name,
+			}
+
+			log.Printf("[INFO] Deleting Lightsail Load Balancer: %s", name)
+			_, err := conn.DeleteLoadBalancer(input)
+
+			if err != nil {
+				sweeperErr := fmt.Errorf("error deleting Lightsail Load Balancer (%s): %s", name, err)
+				log.Printf("[ERROR] %s", sweeperErr)
+				sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+			}
+		}
+
+		if aws.StringValue(output.NextPageToken) == "" {
+			break
+		}
+
+		input.PageToken = output.NextPageToken
+	}
+
+	return sweeperErrs.ErrorOrNil()
+}
+
+func TestAccAWSLightsailLoadBalancer_basic(t *testing.T) {
+	var loadBalancer lightsail.LoadBalancer
+	lightsailLoadBalancerName := fmt.Sprintf("tf-test-lightsail-%d", acctest.RandInt())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t); testAccPreCheckAWSLightsail(t) },
+		IDRefreshName: "aws_lightsail_load_balancer.lightsail_load_balancer_test",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSLightsailLoadBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLightsailLoadBalancerConfig_basic(lightsailLoadBalancerName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSLightsailLoadBalancerExists("aws_lightsail_load_balancer.lightsail_load_balancer_test", &loadBalancer),
```

Comment:
extract `aws_lightsail_load_balancer.lightsail_load_balancer_test` to `resourceName := "aws_lightsail_load_balancer.lightsail_load_balancer_test"` instead of repeating.

it would be also best to use `test` instead of `lightsail_load_balancer_test` to be less verbose and more similar to other acceptance tests in the provider

Reply:
Thanks for the feedback, this change will be implemented once the following pull request is merged by @akonrath https://github.com/akonrath/terraform-provider-aws/pull/96